File: android/app/src/main/kotlin/com/bockmedia/console/ui/testing/BockTestTags.kt
package com.bockmedia.console.ui.testing

/** Stable Compose test tags for instrumented UI / perf audits (debug builds). */
object BockTestTags {
    const val BOTTOM_NAV = "bock_bottom_nav"
    const val NAV_HOME = "bock_nav_home"
    const val NAV_SEARCH = "bock_nav_search"
    const val NAV_LIBRARY = "bock_nav_library"
    const val NAV_DOWNLOADS = "bock_nav_downloads"
    const val NAV_AUTOMATIONS = "bock_nav_automations"

    const val SCREEN_LOADING = "bock_screen_loading"
    const val HOME_FEED = "bock_home_feed"
    const val HOME_GREETING = "bock_home_greeting"
    const val TAB_HEADER = "bock_tab_header"
    const val SEARCH_FIELD = "bock_search_field"
    const val SEARCH_RESULTS = "bock_search_results"
    const val LIBRARY_LIST = "bock_library_list"
    const val AUTOMATIONS_CONTENT = "bock_automations_content"
    const val ANALYTICS_BODY = "bock_analytics_body"
    const val ANALYTICS_TOTAL_PLAYS = "bock_analytics_total_plays"
    const val ACCOUNT_MENU = "bock_account_menu"
    const val ACCOUNT_MENU_BUTTON = "bock_account_menu_button"
    const val SETTINGS_BODY = "bock_settings_body"
    const val SETTINGS_WIFI_ONLY = "bock_settings_wifi_only"
    fun settingsContinue(value: String) = "bock_settings_continue_$value"
    const val FAMILY_ACTING_AS = "bock_family_acting_as"
    const val SPLASH = "bock_splash"
    const val SETUP_SCREEN = "bock_setup_screen"
    const val PROFILE_PICKER = "bock_profile_picker"
}

File: ios/BockMedia/UI/Testing/BockTestTags.swift
import Foundation

/// Stable accessibility identifiers for UI / smoke tests (Debug).
enum BockTestTags {
    static let bottomNav = "bock_bottom_nav"
    static let navHome = "bock_nav_home"
    static let navSearch = "bock_nav_search"
    static let navLibrary = "bock_nav_library"
    static let navAutomations = "bock_nav_automations"

    static let screenLoading = "bock_screen_loading"
    static let homeFeed = "bock_home_feed"
    static let homeGreeting = "bock_home_greeting"
    static let searchField = "bock_search_field"
    static let searchResults = "bock_search_results"
    static let libraryList = "bock_library_list"
    static let automationsContent = "bock_automations_content"
    static let analyticsBody = "bock_analytics_body"
    static let analyticsTotalPlays = "bock_analytics_total_plays"
    static let accountMenuButton = "bock_account_menu_button"
    static let settingsBody = "bock_settings_body"
    static let settingsWifiOnly = "bock_settings_wifi_only"
    static func settingsContinue(_ value: String) -> String { "bock_settings_continue_\(value)" }
    static let profilePicker = "bock_profile_picker"
    static let clientId = "bock_client_id"
    static let setupScreen = "bock_setup_screen"
}
